File: socketClient.go
package main

import (
	"bytes"
	"flag"
	"fmt"
	"log"
	"net"
	"time"
)

var port string

func init() {
	flag.StringVar(&port, "port", "60000", "Specify port to connect")
}

func main() {
	flag.Parse()
	var ADDRESS = net.JoinHostPort("127.0.0.1", port)
	receiveBuf := make([]byte, 10, 16)
	conn, err := net.Dial("tcp4", ADDRESS)
	dealErr(err)
	fmt.Printf("Local network address: %v\n", conn.LocalAddr().String())
	fmt.Printf("Remote network address: %v\n", conn.RemoteAddr().String())
	for {
		numBytesRead, err := conn.Read(receiveBuf)
		dealErr(err)
		fmt.Printf("On %v, read %v bytes: %v\n", time.Now().String(), numBytesRead, string(receiveBuf[:numBytesRead]))
		if bytes.Equal(bytes.ToLower(receiveBuf[:numBytesRead]), []byte("exit")) {
			goto dealExit
		}
		time.Sleep(1 * time.Second)
		numBytesWritten, err := conn.Write([]byte("Happy."))
		dealErr(err)
		fmt.Printf("On %v, written %v bytes.\n", time.Now().String(), numBytesWritten)
	}
dealExit:
	conn.Close()
	fmt.Println("Done.")
}

func dealErr(err error) {
	if err != nil {
		log.Fatal(err.Error())
	}
}

File: socketClient.py
# -*- coding:utf-8 -*-
"""
Dec: socker客户端，接受Go语言server端发送的消息
Created on: 2018.10.11
Author: Iflier
"""

import time
import socket
import argparse
from datetime import datetime


ap = argparse.ArgumentParser()
ap.add_argument("-p", "--port", type=int, default=60000, help="Specify server port to connect")
args = vars(ap.parse_args())

ADDRESS = ("127.0.0.1", args["port"])
sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
sock.connect(ADDRESS)

while True:
    receivedBytes = sock.recv(128)
    print("On {0}, received: {1}".format(datetime.now().strftime("%c"), receivedBytes.decode()))
    if receivedBytes.decode().lower() in ["exit", "quit"]:
        break
    time.sleep(1.0)
    sendBytesNum = sock.send("Happy.".encode())
    print("On {0}, written {1:^5,d} bytes.".format(datetime.now().strftime("%c"), sendBytesNum))

sock.close()
print("Done.")
